### PR TITLE
Ma/appbundle and move

### PR DESCRIPTION
**Pull request by markan**
Rework chef-server-ctl. 
This makes it a gem, appbundles it, and moves it to the src directory alongside the habitat stuff. Previously we were letting the dependencies float free, and that breaks when we upgrade other gems.

This is built on #1513, and will replace it if it goes smoothly.

Will squash this up for final merge.

### CODE REVIEW

File: omnibus/config/software/private-chef-ctl.rb [hunk 1]
```diff
@@ -14,13 +14,16 @@
...
+source path: "#{project.files_path}/../../src/chef-server-ctl"
 
 license :project_license
 skip_transitive_dependency_licensing true
 
+dependency "postgresql96" # for libpq
+
+dependency "appbundler"
 dependency "highline-gem"
 dependency "sequel-gem"
```
Comment (lamont-granquist) [inline omnibus/config/software/private-chef-ctl.rb]: these gems should get moved into the Gemfile.lock to get accurately appbundle'd.

if there are c-libs in the omnibus sources those deps should get moved here.

File: src/chef-server-ctl/lib/chef_server_ctl/helpers/key_ctl_helper.rb [hunk 2]
```diff
@@ -0,0 +1,192 @@
+require "chef/config"
```
Comment (marcparadise) [inline src/chef-server-ctl/lib/chef_server_ctl/helpers/key_ctl_helper.rb]: This looks like a straight move w/ no changes?

File: src/chef-server-ctl/plugins/upgrade.rb [hunk 3]
```diff
@@ -3,11 +3,8 @@
 # All Rights Reserved
 #
 
-begin
-  require_relative 'open_source_chef12_upgrade'
-rescue LoadError
-  require '/opt/opscode/embedded/service/omnibus-ctl/open_source_chef12_upgrade'
-end
+# TODO refactor this for gemified chef-server-ctl
```
Comment (marcparadise) [inline src/chef-server-ctl/plugins/upgrade.rb]: Is this an intentionally left behind TODO?

File: src/chef-server-ctl/lib/chef_server_ctl/helpers/key_ctl_helper.rb (hunk 2 above)
Comment (markan) [reply src/chef-server-ctl/lib/chef_server_ctl/helpers/key_ctl_helper.rb]: Yes; looks like the mv got lost somehow and turned into an add + delete

**Comment by markan** [comment]
oc_erchef is failing in ct. I'm still merging this since master (a5b55c864411fcaea21b7da3fd5646e6c8db306a)  has the same issue, and is broken in other ways that this fixes.

**Pull request merged**